lib/common/pool.c

```diff
@@ -0,0 +1,190 @@
+/**
+ * Copyright (c) 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#include "pool.h"
+#include <stddef.h>  /* size_t */
+#include <stdlib.h>  /* malloc, calloc, free */
+
+#ifdef ZSTD_PTHREAD
+
+#include <pthread.h>
```

Comment:
In latest `zstdmt` branch update,
`threading.h` from Tino's project has been added.
Its objective is to provide a wrapper `pthread` interface for non-POSIX systems. Typically, it needs to support `_WIN32` and non-pthreads systems.

The number of `pthreads` symbols it supports is currently pretty low.
See [`threading.h`](https://github.com/facebook/zstd/blob/zstdmt/lib/common/threading.h) file for more details.